### DISCUSSION TITLE
Make artificially set parameter available from validator

### DISCUSSION THREAD

**Discussion by usr345** [mojolicious/mojo#1856] [Q&A]
I use Mojolicious command for testing a controller function. I artificially set the params for the controller inside a command:

`sub run {
    my ($self, @args) = @_;

    $self->app->controller_class('WWW::Orders');
    my $c = $self->app->build_controller;

    $c->param('name' => 'test');
    $c->process_order();
}
`

Controller function looks like this:

`sub process_order {
    my $c = shift;

    my $v = $c->validation;
    print Dumper($v->param('name')); # prints undef
    print Dumper($c->param('name')); # prints "test"
`

Is it possible to set the parameter so it was available from the validator?

**Comment by Grinnz** [comment]
Controller actions are normally tested using [Test::Mojo](https://metacpan.org/pod/Test::Mojo) or manually with [Mojolicious::Command::get](https://metacpan.org/pod/Mojolicious::Command::get), in either case loaded from the application and passed the route to the action.

**Reply by usr345** [reply]
I understand that my case is not a usual one. I have 2 databases: one is my small personal database and another is a big development one used by other programmers. I am trying to write closed tests that will use only my small database. But I can't write a GET request cause it will require copying a lot of extra tables from the main database to process.

I found that I can set parameters for the controller. Validator parameters is the only thing left.

**Comment by daleif** [comment]
You have direct access to the validation parameters. See <https://docs.mojolicious.org/Mojolicious/Validator/Validation> `input` and `output`. I sometimes use this to give optional elements a default value if they we not chosen, or to overwrite a user choice.
